x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -267,12 +273,19 @@ protected void masterOperation(
                                                         parentTask,
                                                         ActionListener.wrap(
                                                             mergeIndexResp -> listener.onResponse(AcknowledgedResponse.TRUE),
-                                                            e -> listener.onFailure(
-                                                                new ElasticsearchException(
-                                                                    "Failed to force-merge index [" + rollupIndexName + "]",
+                                                            e -> {
+                                                                /*
+                                                                 * At this point rollup has been successful even if force-merge fails.
+                                                                 * Also, we have deleted the source index and there is no way we can
+                                                                 * roll back and restart the operation. So, we should not fail the rollup
+                                                                 * operation.
+                                                                 */
+                                                                logger.error(
+                                                                    "Failed to force-merge rollup index [" + rollupIndexName + "]",
```

Comment:
:+1:

It's probably worth a test that fails here and makes sure we don't fall over. I think this will do the job but it's probably worth tracing this path. Both as a "for now" thing and to make sure that changes in the future don't cause us to lose this property.